packages/dids/src/dht.ts

```diff
@@ -0,0 +1,312 @@
+import dns, {AUTHORITATIVE_ANSWER, Packet, TxtAnswer} from 'dns-packet';
+import Encoder from '@decentralized-identity/ion-sdk/dist/lib/Encoder.js';
+import {Pkarr, SignedPacket, z32} from 'pkarr';
+import type {PublicKeyJwk, Web5Crypto} from '@web5/crypto';
+import {Jose} from '@web5/crypto';
+import type {DidDocument} from './types.js';
+
+const PKARR_RELAYS = ['https://diddht.tbddev.org'];
+const TTL = 7200;
+
+export class DidDht {
+
+  /**
+     * Publishes a DID Document to the DHT
+     * @param keypair The keypair to sign the document with
+     * @param did The DID Document to publish
+     */
+  public static async publishDidDocument(keypair: Web5Crypto.CryptoKeyPair, did: DidDocument): Promise<boolean> {
+    const packet = await DidDht.toDnsPacket(did);
+    const pkarrKeypair = {
+      publicKey : keypair.publicKey.material,
+      secretKey : new Uint8Array([...keypair.privateKey.material, ...keypair.publicKey.material])
+    };
+    const signedPacket = SignedPacket.fromPacket(pkarrKeypair, packet);
+    const results = await Promise.all(PKARR_RELAYS.map(relay => Pkarr.relayPut(relay, signedPacket)));
```

Comment:
@decentralgabe should the pkarr relay used to publish be a configurable parameter?  Fine with addressing this in a future PR.

Reply:
I believe it's something the library should take a stance on, but may be able to override

Reply:
updated to support a configurable relay with a default set by the lib